app/src/handlers/aggregate/global/GlobalPlayerAggregate.ts

```diff
@@ -0,0 +1,104 @@
+        this.factory = factory;
+    }
+
+    public async handle(event: DeathEvent): Promise<boolean> {
+        GlobalPlayerAggregate.logger.debug('GlobalPlayerAggregate.handle');
+
+        // Check both attacker and victim for existence
+        const checks = [event.characterId, event.attackerCharacterId];
+
+        for (const id of checks) {
+            // Create initial record if doesn't exist
+            if (!await this.factory.model.exists({
+                player: id,
+            })) {
+                await this.insertInitial(event, id);
+            }
+        }
+
+        const attackerDocs = [];
+        const victimDocs = [];
+
+        // Victim deaths always counted in every case
+        victimDocs.push({$inc: {deaths: 1}});
+
+        if (!event.isTeamkill && !event.isSuicide) {
+            attackerDocs.push({$inc: {kills: 1}});
+        }
+
+        if (event.isTeamkill) {
+            attackerDocs.push({$inc: {teamKills: 1}});
+        }
+
+        if (event.isSuicide) {
+            // Attacker and victim are the same here, so it doesn't matter which
+            victimDocs.push({$inc: {suicides: 1}});
+        }
+
+        if (event.isHeadshot) {
+            attackerDocs.push({$inc: {headshots: 1}});
+        }
+
+        // It's an old promise sir, but it checks out (tried Async, doesn't work with forEach)
+        attackerDocs.forEach((doc) => {
+            void this.factory.model.updateOne(
+                {player: event.attackerCharacterId},
+                doc,
+            ).catch((err) => {
+                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                GlobalPlayerAggregate.logger.error(`Updating GlobalPlayerAggregate Attacker Error! ${err}`);
+            });
+        });
+
+        victimDocs.forEach((doc) => {
+            void this.factory.model.updateOne(
+                {player: event.characterId},
+                doc,
+            ).catch((err) => {
+                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                GlobalPlayerAggregate.logger.error(`Updating GlobalPlayerAggregate Victim Error! ${err}`);
+            });
+        });
+
+        return true;
+    }
+
+    private async insertInitial(event: DeathEvent, characterId: string): Promise<boolean> {
+        GlobalPlayerAggregate.logger.debug(`Adding Initial AlertPlayerAggregate Record for C: ${characterId} | A: ${event.alert.alertId}`);
+
+        const player = {
+            player: characterId,
+            world: event.world,
+            kills: 0,
+            deaths: 0,
+            teamKills: 0,
+            suicides: 0,
+            headshots: 0,
+        };
+
+        try {
+            const row = await this.factory.saveDocument(player);
+            GlobalPlayerAggregate.logger.debug(`Inserted initial GlobalPlayerAggregate record for C: ${row.player}`);
```

Comment:
`for C:` ?

Reply:
Character, probably could name it better